src/AppRoot.tsx

```diff
@@ -497,11 +497,40 @@ class AppRoot extends React.Component<Props, State> {
         }
       )
     }
+
+    /* istanbul ignore else */
+    if (!this.onDeviceChangeListener) {
+      const { hardware } = this.props
+      ;(async () => {
+        // get initial status
+        const accessibleControllerStatus = await hardware.readAccesssibleControllerStatus()
+
+        this.setState({
+          hasAccessibleControllerAttached: accessibleControllerStatus.connected,
+        })
+
+        // watch for changes
+        this.onDeviceChangeListener = hardware.onDeviceChange.add(
+          (changeType, device) => {
+            /* istanbul ignore else */
+            if (isAccessibleController(device)) {
+              this.setState({
+                hasAccessibleControllerAttached:
+                  changeType === 0 /* ChangeType.Add */,
```

Comment:
I can’t use the actual enum value here because it doesn’t exist as a value, only a type. It’s defined in a TypeScript declaration file. I should fix that, but maybe not in this commit.

Reply:
Yeah, I'm going to leave this for later.